functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,182 @@
+STRATA_DEPOSIT_LOWER_BOUND = ((ROLLUP_DEPOSIT_AMOUNT // 1.01) * SATS_TO_WEI) - (
+    2 * GAS * GWEI_TO_WEI
+)
+
+
+@flexitest.register
+class BridgeWithdrawHappyTest(flexitest.Test):
+    """
+    Makes two DRT deposits to the same EL address, then makes a withdrawal to a change address.
+
+    Checks if the balance of the EL address is expected
+    and if the BTC balance of the change address is expected.
+    """
+
+    def __init__(self, ctx: flexitest.InitContext):
+        ctx.set_env(BasicEnvConfig(pre_generate_blocks=101))
+        self.logger = get_logger("BridgeWithdrawHappyTest")
+
+    def main(self, ctx: flexitest.RunContext):
+        address = get_address(0)
+        withdraw_address = get_address(1)
+        self.logger.debug(f"Address: {address}")
+        self.logger.debug(f"Change Address: {withdraw_address}")
+        self.logger.debug(f"Gas: {GAS}")
+
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        reth = ctx.get_service("reth")
+
+        seqrpc = seq.create_rpc()
+        btcrpc: BitcoindClient = btc.create_rpc()
+        rethrpc = reth.create_rpc()
+
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+        el_address = web3.address
+        self.logger.debug(f"EL address: {el_address}")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        bridge_pk = get_bridge_pubkey(seqrpc)
+
+        # Generate Plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Deposit to the EL address
+        # NOTE: we need 2 deposits to make sure we have funds for gas
+        self.make_drt(ctx, el_address, bridge_pk)
+        self.make_drt(ctx, el_address, bridge_pk)
+        time.sleep(0.5)
+
+        # Send funds to the bridge precompile address for a withdrawal
+        change_address_pk = extract_p2tr_pubkey(withdraw_address)
+        self.logger.debug(f"Change Address PK: {change_address_pk}")
+        l2_tx_hash = self.make_withdraw(ctx, change_address_pk, GAS).hex()
+        self.logger.debug(f"Sent withdrawal transaction with hash: {l2_tx_hash}")
+
+        # Wait for the withdrawal to be processed
+        time.sleep(DEFAULT_BLOCK_TIME_SEC * 2)
+
+        l2_tx_state = web3.eth.get_transaction_receipt(l2_tx_hash)
+        self.logger.debug(f"Transaction state (web3): {l2_tx_state}")
+
+        # Make sure that the balance is zero
+        balance = int(rethrpc.eth_getBalance(el_address), 16)
```

Comment:
This should just check if the difference in balance before and after the withdrawal equals the `DEPOSIT_DENOMINATION` - `gas`. Doing it this way makes the tests a bit flaky (for example, if we want to test multiple withdrawals).